### DISCUSSION TITLE
Is there a way to add widgets programmatically?

### DISCUSSION THREAD

**Discussion by klott** [Textualize/textual#3170] [Q&A] [answered]
I'm not sure if I'm just having a moment or if this simply isn't possible...
I'm trying to have a multi-select that, on select, spawns a text input field per select. Here's a wireframe:
![image](https://github.com/Textualize/textual/assets/15988992/1d8c6346-7569-4f0c-913a-24560d731c0b)

In my mind, adding new widgets in `@on(SelectionList.SelectedChanged` would work but what I can't figure out is how to add and render new widgets programmatically outside of the `compose()` method. Am I just approaching this completely backwards? Any tips?

**Comment by davep** [comment, marked as answer]
The Textual tutorial covers this: https://textual.textualize.io/tutorial/#dynamic-widgets

**Reply by klott** [reply]
ah, I read the Guide cover to cover but haven't revisited the tutorial since the beginning. solution looks like:
```
new_widget = WidgetClass()
self.query_one("#id").mount(new_widget)
```

**Comment by willmcgugan** [comment]
Nice wireframe 👌

**Reply by klott** [reply]
I was thrilled to see excalidraw in the documentation. Love the tool! Cheers for creating such a well-designed library <3
